cni/network/invoker_cns.go

```diff
@@ -151,18 +151,35 @@ func setHostOptions(nwCfg *cni.NetworkConfig, hostSubnetPrefix *net.IPNet, ncSub
 		},
 	}
 
-	azureDNSMatch := fmt.Sprintf(" -m addrtype ! --dst-type local -s %s -d %s -p %s --dport %d", ncSubnetPrefix.String(), iptables.AzureDNS, iptables.UDP, iptables.DNSPort)
-
+	ncSubnetPrefixStr := ncSubnetPrefix.String()
+	azureDNSMatch := fmt.Sprintf(" -m addrtype ! --dst-type local -s %s -d %s -p %s --dport %d", ncSubnetPrefixStr, iptables.AzureDNS, iptables.UDP, iptables.DNSPort)
+	snatPrimaryIPJump := fmt.Sprintf("%s --to %s", iptables.Snat, info.ncPrimaryIP)
 	// TODO remove this rule once we remove adding MASQUEARDE from AgentBaker, check below PR
 	// https://github.com/Azure/AgentBaker/pull/367/files
-	podTrafficAccept := fmt.Sprintf(" -m iprange  ! --dst-range 168.63.129.16-168.63.129.16  -s %s ", ncSubnetPrefix.String())
-	snatPrimaryIPJump := fmt.Sprintf("%s --to %s", iptables.Snat, info.ncPrimaryIP)
-	options[network.IPTablesKey] = []iptables.IPTableEntry{
+	podTrafficAccept := fmt.Sprintf(" -m iprange  ! --dst-range 168.63.129.16-168.63.129.16  -s %s ", ncSubnetPrefixStr)
+	options[network.IPTablesKey] = make([]iptables.IPTableEntry, 0, 6)
+
+	if nwCfg.ExecutionMode == string(NonOverlappingMultitenancy) && nwCfg.NonOverlappingMultitenancyMode.ConnectivityType == string(Snat) {
+		if nwCfg.NonOverlappingMultitenancyMode.EnablePodToNodeSnat {
```

Comment:
Can we please update the description of this PR with what new rules we are adding, where we are adding and for what conditions. so that tomorrow it will be very easy for us to just look up the description and not read the code and understand.